tests/components/application-connector/test/compass-runtime-agent/secret.go

```diff
@@ -0,0 +1,116 @@
+package compass_runtime_agent
+
+import (
+	"context"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	restclient "k8s.io/client-go/rest"
+	"reflect"
+	"testing"
+)
+
+type secretData struct {
+	password     string
+	username     string
+	crt          string
+	key          string
+	clientId     string
+	clientSecret string
+}
+
+type Client interface {
+	GetSecret() (secretData, error)
+}
+
+type client struct {
+	secretsClient v1.SecretInterface
+	secretName    string
+}
+
+func NewClient(secrets v1.SecretInterface, secretName string) Client {
+	return &client{
+		secretsClient: secrets,
+		secretName:    secretName,
+	}
+}
+
+func newSecretsInterface(namespace string) (v1.SecretInterface, error) {
+	k8sConfig, err := restclient.InClusterConfig()
+	if err != nil {
+		logrus.Warnf("Failed to read in cluster config: %s", err.Error())
+		logrus.Info("Trying to initialize with local config")
+		return nil, errors.Wrap(err, "Failed to read in cluster config")
+	}
+
+	coreClientset, err := kubernetes.NewForConfig(k8sConfig)
+	if err != nil {
+		return nil, errors.Errorf("failed to create k8s core client, %s", err.Error())
+	}
+
+	return coreClientset.CoreV1().Secrets(namespace), nil
+}
+
+func initalizeSecretInterface(namespace string) (v1.SecretInterface, error) {
+	secretsRepo, err := newSecretsInterface(namespace)
```

Comment:
IMHO there is no need to have a separate function here. You can combine both into a single one.